### ISSUE TITLE
BSL error stepping big-bang

### ISSUE THREAD

**Issue by maueroats** [racket/htdp#233]
I know this occurred running the stepper on a `big-bang`, which should not be done.

### Error message
This occurs in the stepper.

```
skip-step?: expects (or/c (listof procedure?) #false) or false, given (list function "supposed to be skipped" function)
  in: the 2nd argument of
      (->
       (or/c
        'normal-break
        'normal-break/values
        'result-exp-break
        'result-value-break
        'double-break
        'late-let-break
        'expr-finished-break
        'define-struct-break)
       (or/c (listof procedure?) #false)
       render-settings?
       boolean?)
  contract from: 
      <pkgs>/htdp-lib/stepper/private/reconstruct.rkt
  blaming: <pkgs>/htdp-lib/stepper/private/model.rkt
   (assuming the contract is correct)
  at: <pkgs>/htdp-lib/stepper/private/reconstruct.rkt:48:2
```

## Source code

The complaint appears to be originating from [this line](https://github.com/racket/htdp/blob/aef9ae1c8e4c1e256edde5ab842d20d558202ee2/htdp-lib/stepper/private/annotate.rkt#L678). The complaint looks like an error that the string `"supposed to be skipped"` does not satisfy `procedure?`.

This might arise from more careful enforcement of a contract than in the past?

Would replacing the string with `(lambda () "supposed to be skipped")` cause problems?

## Reproduce

```scheme
(require 2htdp/image)
(require 2htdp/universe)

(define nothingness (square 0 "solid" "transparent"))

(define (2draw3 img)
  (overlay img (empty-scene 400 100)))

(big-bang nothingness
  (to-draw 2draw3))
```

**Comment by shhyou** [comment]
It looks like d53a1978acd1 changed `run-it` to `send` which could be the cause.

**Comment by rfindler** [comment]
I wonder if using quasisyntax/loc would work but on the call to `run-it` (perhaps putting it into a non-tail position too?)

**Comment by shhyou** [comment]
Well, just using the previous version of `run-it` is not sufficient, but it has to be the commented out `run-it` function in 36fc53949a8.

@mfelleisen in <https://github.com/racket/htdp/commit/36fc53949a8bd3df5ffca2ecce3f3a3b2b078a63>, why was the old `run-it` replaced by only the `send` part? I can use `raco exe` to create executables with the commented out `run-it`, so is it okay to uncomment `run-it`?
```diff
  ;; (-> Object) -> Any
+ (define (run-it o)
+   (send (o) last))
+ 
+ #;
  (define (run-it o)
    (define esp (make-eventspace))
    (define thd (eventspace-handler-thread esp))
    (with-handlers ((exn:break? (lambda (x) (break-thread thd))))
      ...
```

**Comment by rfindler** [comment]
I think the highlighting (the reason behind https://github.com/racket/htdp/commit/d53a1978acd16f16c29746b9b68e096be955377b) is affected by things like tail position (which the run it that was replaced in https://github.com/racket/htdp/commit/36fc53949a8bd3df5ffca2ecce3f3a3b2b078a63 would have had) and the stepper can be affected by syntax properties or annotations that say things like "please ignore functions named `run-it`". 

The function that creates a new eventspace and disables breaks, etc, seems like a concern unrelated to the stepper and also unrelated to highlighting.

**Comment by shhyou** [comment]
> I think the highlighting (the reason behind [d53a197](https://github.com/racket/htdp/commit/d53a1978acd16f16c29746b9b68e096be955377b)) is affected by things like tail position (which the run it that was replaced in [36fc539](https://github.com/racket/htdp/commit/36fc53949a8bd3df5ffca2ecce3f3a3b2b078a63) would have had differently) and the stepper can be affected by syntax properties or annotations that say things like "please ignore functions named `run-it`".
> 

Yes, I see the point behind d53a1978ac that fixes the source location for highlighting, but it turns out that only reverting the inlining of `run-it` and using `run-it` per the `(define (run-it o) (send (o) last))` code does not solve the problem. But it turns out that using the commented-out `run-it` function from 36fc53949a8 does solve the bug, so something else must be going on.

> The function that creates a new eventspace and disables breaks, etc, seems like a concern unrelated to the stepper and also unrelated to highlighting.

The weird thing is using the `run-it` that creates an eventspace does make a difference and the stepper doesn't break anymore. It also runs the object creation (`(o)`) in a different thread, so some of these must be affecting the continuation marks.

**Comment by mfelleisen** [comment]
This arrangement was carefully "negotiated" between John and myself -- but of course, the rationale is in two heads and not written down. @shhyou is correct in that the thread arranges continuation marks differently and that's the key to the stepper marking up the stack.
